laiaaa0/assignment3/TestingFunctions.cpp

```diff
@@ -0,0 +1,104 @@
+#include "codeu_test_lib.h"
+#include "Dictionary.h"
+void CreateSampleGrid(std::vector<std::vector<char> > &grid){
+    grid.clear();
+
+    std::vector<char> line1{'a','a','r'};
+    std::vector<char> line2{'t','c','d'};
+    grid.push_back(line1);
+    grid.push_back(line2);
+
+}
+
+
+Dictionary * test_CreateDictionaryFromVector(){
+
+    std::vector<std::string> v{"car","card","cart","cat","rat"};
+    Dictionary * d = new Dictionary(v);
+    return d;
+
+}
+void test_IsWordInDictionary(Dictionary * d){
+  EXPECT_TRUE(d->IsWord("cat"));
+  EXPECT_TRUE(d->IsWord("car"));
+  EXPECT_TRUE(d->IsWord("rat"));
+  EXPECT_FALSE(d->IsWord("c"));
+  EXPECT_FALSE(d->IsWord("RAT"));
+  EXPECT_FALSE(d->IsWord("ca"));
+
+}
+void test_IsPrefixInDictionary (Dictionary * d){
+  EXPECT_TRUE(d->IsPrefix("c"));
+  EXPECT_TRUE(d->IsPrefix("ca"));
+  EXPECT_TRUE(d->IsPrefix("cat"));
+  EXPECT_TRUE(d->IsPrefix("card"));
+  EXPECT_FALSE(d->IsPrefix(""));
```

Comment:
This is not obvious from the intuitive definition of prefix.
But it is a fine choice to make. In that case, however, I would document this where `IsPrefix()` is defined.